cmd/update.go

```diff
@@ -0,0 +1,496 @@
+			}
+		}
+		return errors.New("binary not found in tar.gz")
+
+	default: // raw binary
+		f, err := os.Open(assetPath)
+		if err != nil {
+			return err
+		}
+		defer f.Close()
+		return replaceExecutable(binPath, f, 0)
+	}
+}
+
+// Checksum verification
+
+func verifySHA256(filePath, shaPath string) error {
+	info, err := os.Stat(shaPath)
+	if err != nil {
+		return err
+	}
+	if info.Size() > maxChecksumSize {
+		return fmt.Errorf("checksum file too large: %d bytes", info.Size())
+	}
+
+	f, err := os.Open(shaPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	scanner := bufio.NewScanner(f)
+	if !scanner.Scan() {
+		return errors.New("empty sha256 file")
+	}
+	hashHex := strings.Fields(scanner.Text())[0] // ignore trailing filename
+	want, err := hex.DecodeString(hashHex)
+	if err != nil || len(want) != sha256.Size {
+		return errors.New("invalid sha256 checksum")
+	}
+
+	bin, err := os.Open(filePath)
+	if err != nil {
+		return err
+	}
+	defer bin.Close()
+
+	h := sha256.New()
+	if _, err = io.Copy(h, bin); err != nil {
+		return err
+	}
+	got := h.Sum(nil)
+
+	if !bytes.Equal(got, want) {
+		return fmt.Errorf("checksum mismatch: expected %s got %s",
+			hashHex, hex.EncodeToString(got))
+	}
+	return nil
+}
+
+// Cobra command
+
+var updateCmd = &cobra.Command{
+	Use:   "update",
+	Short: "Download and install the newest Oasis CLI release",
+	Long: `Checks GitHub releases for a newer Oasis CLI version that matches
+your platform, downloads it and replaces the current binary.`,
+	Args: cobra.NoArgs,
+	Run: func(_ *cobra.Command, _ []string) {
+		// 0 – finalise any staged update (Windows)
+		exePath, err := osExecutable()
+		cobra.CheckErr(err)
+		finishPendingWindowsUpdate(exePath)
+
+		// 1 – query GitHub
+		apiCtx, cancel := context.WithTimeout(context.Background(), apiTimeout)
+		defer cancel()
+
+		fmt.Println("Checking for updates...")
+		rel, err := fetchLatest(apiCtx)
+		cobra.CheckErr(err)
+
+		current := cliVersion.Software
+		latest := rel.TagName
+
+		isNewer, err := semverGreater(latest, current)
+		cobra.CheckErr(err)
+		if !isNewer {
+			fmt.Printf("oasis CLI is already up to date (%s).\n", current)
+			return
+		}
+
+		assetName, assetURL, err := findAsset(rel)
+		cobra.CheckErr(err)
+		fmt.Printf("New version %s available (%s).\n", latest, assetName)
+
+		// 2 – download asset
+		dlCtx, cancelDL := context.WithTimeout(context.Background(), downloadTimeout)
+		defer cancelDL()
+
+		assetPath, err := downloadFunc(dlCtx, assetURL)
+		cobra.CheckErr(err)
+		defer func() { _ = os.Remove(assetPath) }()
+
+		// 3 – checksum
+		if shaURL, ok := findSha256For(rel, assetName); ok {
+			if shaPath, err2 := downloadFunc(dlCtx, shaURL); err2 == nil {
+				defer func() { _ = os.Remove(shaPath) }()
+				cobra.CheckErr(verifySHA256(assetPath, shaPath))
+				fmt.Println("Checksum verified OK")
+			} else {
+				fmt.Printf("WARNING: couldn't download checksum file: %v (continuing)\n", err2)
```

Comment:
IMO this should be fatal.

```suggestion
				cobra.CheckErr(fmt.Printf("WARNING: couldn't download checksum file: %v\n", err2))
```